### PR TITLE
Adding changes to send invalid client when signature validation fails in jwks based validation

### PR DESCRIPTION
**Pull request by SachiniSiriwardene**
## Automation tests
 - Unit tests 
   > Code coverage information
 - Integration tests
   > Details about the test cases and coverage

## Security checks
 - Followed secure coding standards in http://wso2.com/technical-reports/wso2-secure-engineering-guidelines? yes/no
 - Ran FindSecurityBugs plugin and verified report? yes/no
 - Confirmed that this PR doesn't commit any keys, passwords, tokens, usernames, or other secrets? yes/no

### CODE REVIEW

**Comment by DMHP** [comment]
@SachiniSiriwardene , In the public Jira please put a comment saying that this fix contains a behavioural change and mention that we need to add this details to IS 5.9.0 migration doc. Hence please keep the Jira opened.

File: component/client-handler/src/main/java/org/wso2/carbon/identity/oauth2/token/handler/clientauth/jwt/validator/JWTValidator.java [hunk 1]
```diff
@@ -399,8 +399,9 @@ private boolean isValidSignature(String clientId, SignedJWT signedJWT, String te
                     isValidSignature = new JWKSBasedJWTValidator().validateSignature(jwtString, jwksUri, alg, options);
                 }
             } catch (IdentityOAuth2Exception e) {
-                String errorMessage = "Error occurred while validating signature using jwks";
-                throw new OAuthClientAuthnException(errorMessage, OAuth2ErrorCodes.INVALID_REQUEST, e);
+                String errorMessage = "Error occurred while validating signature using jwks ";
+                log.error(errorMessage, e);
+                return false;
```
Comment (DMHP) [inline component/client-handler/src/main/java/org/wso2/carbon/identity/oauth2/token/handler/clientauth/jwt/validator/JWTValidator.java]: Without returning false lets throw the exception from this level. throw new OAuthClientAuthnException(errorMessage, OAuth2ErrorCodes.INVALID_CLIENT, e);

Comment (SachiniSiriwardene) [reply component/client-handler/src/main/java/org/wso2/carbon/identity/oauth2/token/handler/clientauth/jwt/validator/JWTValidator.java]: if we throw the exception, the correct error code is not sent. That is why this was changed

File: component/client-handler/src/main/java/org/wso2/carbon/identity/oauth2/token/handler/clientauth/jwt/validator/JWTValidator.java [hunk 2]
```diff
@@ -399,8 +399,9 @@ private boolean isValidSignature(String clientId, SignedJWT signedJWT, String te
                     isValidSignature = new JWKSBasedJWTValidator().validateSignature(jwtString, jwksUri, alg, options);
                 }
             } catch (IdentityOAuth2Exception e) {
-                String errorMessage = "Error occurred while validating signature using jwks";
-                throw new OAuthClientAuthnException(errorMessage, OAuth2ErrorCodes.INVALID_REQUEST, e);
+                String errorMessage = "Error occurred while validating signature using jwks ";
```
Comment (kasundharmadasa) [inline component/client-handler/src/main/java/org/wso2/carbon/identity/oauth2/token/handler/clientauth/jwt/validator/JWTValidator.java]: Remove the space after "jwks"

Comment (SachiniSiriwardene) [reply component/client-handler/src/main/java/org/wso2/carbon/identity/oauth2/token/handler/clientauth/jwt/validator/JWTValidator.java]: fixed in https://github.com/wso2-extensions/identity-oauth-addons/commit/8ba88067e171f3bbaa355f2f2f503585d4fb428c

**Pull request merged**